cmd/api_key/api_key.go

```diff
@@ -0,0 +1,227 @@
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		name, _ := cmd.Flags().GetString("name")
+		duration, _ := cmd.Flags().GetInt32("duration")
+
+		unit, _ := cmd.Flags().GetString("unit")
+		var expiryHours int
+		validTimeUnit := false
+		for k, v := range GetKeyTimeConversionMap() {
+			if strings.ToUpper(unit) == k {
+				validTimeUnit = true
+				expiryHours = int(duration) * v
+			}
+		}
+		if !validTimeUnit {
+			logrus.Fatalln("Only Hours, Days, and Months time units are allowed.")
+		}
+
+		apiKeySpec, err := authApi.CreateApiKeySpec(name, expiryHours)
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		if cmd.Flags().Changed("description") {
+			description, _ := cmd.Flags().GetString("description")
+			apiKeySpec.SetDescription(description)
+		}
+
+		if cmd.Flags().Changed("role-name") {
+			roleName, _ := cmd.Flags().GetString("role-name")
+			roleId, err := authApi.GetRoleIdByName(roleName)
+			if err != nil {
+				logrus.Fatal(err)
+			}
+			apiKeySpec.SetRoleId(roleId)
+		}
+
+		resp, r, err := authApi.CreateApiKey().ApiKeySpec(*apiKeySpec).Execute()
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		apiKeyCtx := formatter.Context{
+			Output: os.Stdout,
+			Format: formatter.NewApiKeyFormat(viper.GetString("output")),
+		}
+
+		formatter.SingleApiKeyWrite(apiKeyCtx, resp.GetData())
+
+		fmt.Printf("\nAPI Key: %s \n", formatter.Colorize(resp.GetJwt(), formatter.GREEN_COLOR))
+		fmt.Printf("\nThe API key is only shown once after creation. Copy and store it securely.\n")
+	},
+}
+
+var revokeApiKeyCmd = &cobra.Command{
+	Use:   "revoke",
+	Short: "Revoke an API Key",
+	Long:  "Revoke an API Key",
+	PreRun: func(cmd *cobra.Command, args []string) {
+		viper.BindPFlag("force", cmd.Flags().Lookup("force"))
+		name, _ := cmd.Flags().GetString("name")
+		err := util.ConfirmCommand(fmt.Sprintf("Are you sure you want to revoke the %s: %s", "API Key", name), viper.GetBool("force"))
+		if err != nil {
+			logrus.Fatal(err)
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		name, _ := cmd.Flags().GetString("name")
+		keyId, err := authApi.GetKeyIdByName(name)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		response, err := authApi.RevokeApiKey(keyId).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", response)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		fmt.Printf("The API key %s has been successfully revoked.\n", formatter.Colorize(name, formatter.GREEN_COLOR))
+	},
+}
+
+func init() {
+	ApiKeyCmd.AddCommand(listApiKeysCmd)
+	listApiKeysCmd.Flags().String("name", "", "[OPTIONAL] To filter by API Key name.")
```

Comment:
required should be first .. so add required in top then add other in alph. order see https://github.com/yugabyte/ybm-cli/blob/main/cmd/cluster/create_cluster.go#L167